### PR TITLE
mat: update docs for SVD.SolveTo and SVD.SolveVecTo

### PR DESCRIPTION
**Pull request by vladimir-ch**
This updates the documentation as promised in #1316.

Please take a look.

### CODE REVIEW

File: mat/svd.go [hunk 1]
```diff
@@ -272,13 +272,23 @@ func (svd *SVD) VTo(dst *Dense) {
...
-// AX = B, using a rank-truncated SVD. The result is placed in dst.
-// It returns the residuals calculated from the complete SVD. For this value to be
-// valid the factorization must have been performed with at least SVDFullU.
-// The decomposition must have been factorized computing both the U and V singular
-// vectors, and dst must be either empty or have a row dimension equal to rank.
-// The supplied rank must be a positive end-slice index into svd.Values.
+// SolveTo calculates the minimum-norm solution to a linear least squares problem
+//  minimize over n-element vectors x: |b - A*x|_2 and |x|_2
+// where b is a given m-element vector, using the SVD of m×n matrix A stored in
+// the receiver. A may be rank-deficient, that is, the given effective rank can be
```
Comment (vladimir-ch) [inline mat/svd.go:278]: Should the text suggest explicitly using SVD.Rank?

Comment (kortschak) [reply mat/svd.go:278]: I think it probably should.

Comment (vladimir-ch) [reply mat/svd.go:278]: Done.

**Comment by vladimir-ch** [comment]
PTAL

**Pull request merged**